### PR TITLE
feat(cloud-megamenu): megamenu tabbing functionality foundation

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[Cloud Masthead] Setup baseline component for the new Mega Menu dropdown design #5741

### Description

This PR is to set up the foundation of the Cloud Megamenu which uses a tabbing functionality.

HC team will handle the styling and detail work.

### Changelog

**New**

- `cloud-megamenu-tab` & `cloud-megamenu-tabs` components extending from carbon-web-component tab component
- various components for styling

**Changed**

- adds the `description` field to the `BasicLink` interface in translation types for `services-store` package

### CODE REVIEW

**Comment by annawen1** [comment]
Hi @proeung, here is the PR for the cloud megamenu tabbing functionality. 

CC: @huntermacd

File: packages/web-components/package.json [hunk 1]
```diff
@@ -82,6 +82,7 @@
     "@carbon/layout": "10.19.0",
     "carbon-components": "10.29.0",
     "carbon-web-components": "1.13.0",
+    "js-cookie": "^2.2.1",
```
Comment (jeffchew) [inline packages/web-components/package.json:85]: @annawen1 would you be able to set up the cookie utility in this PR and move into that package?

Comment (annawen1) [reply packages/web-components/package.json:85]: This is resolved in Hunter's PR: feat(cloud-masthead): add Cloud-specific container and composite components #5593

**Review by proeung — APPROVED** [review]
@annawen1 Looks great overall! Thanks again for implementing the tabbing functionality and extending the components that are needed for the cloud variant of the mega menu. Awesome work!

@annawen1 @jeffchew I would assume that we'll need to get this PR (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5593) merged down first before we get this one in. Also, I'll follow up with a new PR that addresses the styling and polishing work that's remaining for the mega menu dropdown.

**Comment by annawen1** [comment]
@shixiedesign design review not needed at the moment as this is still a work in progress - wrapped behind a feature flag of course. This will not affect our existing masthead or megamenu

**Review by andysherman2121 — APPROVED** [review]
Looks good to me! thanks anna

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

**Pull request merged**